src/FsAutoComplete.Core/Workspace.fs

```diff
@@ -0,0 +1,164 @@
+module FsAutoComplete.WorkspacePeek
+
+open System
+open System.IO
+
+type SolutionData = {
+    Items: SolutionItem list
+    Configurations: SolutionConfiguration list
+    }
+and SolutionConfiguration = {
+    Id: string
+    ConfigurationName: string
+    PlatformName: string
+    IncludeInBuild: bool
+    }
+and SolutionItem = {
+    Guid: Guid
+    Name: string
+    Kind: SolutionItemKind
+    }
+and SolutionItemKind =
+    | MsbuildFormat of SolutionItemMsbuildConfiguration list
+    | Folder of (SolutionItem list) * (string list)
+    | Unsupported
+    | Unknown
+and SolutionItemMsbuildConfiguration = {
+    Id: string
+    ConfigurationName: string
+    PlatformName: string
+    }
+
+[<RequireQualifiedAccess>]
+type Interesting =
```

Comment:
suggestions? cannot think anything else atm.
this is the list of stuff who can or cannot be useful in the dir (for example fsx are not returned by command later)